runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -368,17 +365,11 @@ public void insert(@ConflictParameter K key, V value) {
         // operations are needed to be executed on the internal data structure
     }
 
-    /**
-     * Returns a filtered {@link List} view of the values contained in this map.
-     * This method has a memory/CPU advantage over the map iterators as no deep copy
-     * is actually performed.
-     *
-     * @param valuePredicate java predicate (function to evaluate)
-     * @return a view of the values contained in this map meeting the predicate condition.
-     */
+    /** {@inheritDoc} */
+    @Override
     @Accessor
     public List<V> scanAndFilter(Predicate<? super V> valuePredicate) {
-        return pool.submit(() -> mainMap.entryStream()
+        return pool.submit(() -> mainMap.mutableEntryStream()
                 .map(Entry::getValue).filter(valuePredicate)
```

Comment:
Can you use try-resources on the stream so that It is explicitly closed after its used.

Like this 

https://github.com/CorfuDB/CorfuDB/commit/e8f154e9330bf164003147b16c50fd77c6df58a5#diff-6f23471c25711fa551fa3b560d27d49dR154

Reply:
done.